src/hotspot/share/gc/g1/g1CollectedHeap.inline.hpp

```diff
@@ -111,6 +111,17 @@ inline HeapRegion* G1CollectedHeap::next_region_in_humongous(HeapRegion* hr) con
   return _hrm.next_region_in_humongous(hr);
 }
 
+template <typename Functor>
+inline void G1CollectedHeap::humongous_obj_regions_iterate(HeapRegion* start, Functor&& f) {
```

Comment:
Forwarding reference might be overkill for this case, `const Functor&` should be just fine.